src/main/scala/mesosphere/marathon/api/v2/Validation.scala

```diff
@@ -92,6 +93,17 @@ object Validation {
     }
   }
 
+  def fetchUriIsValid: Validator[FetchUri] = {
+    new Validator[FetchUri] {
+      def apply(uri: FetchUri) = {
+        Try {
+          new URL(uri.uri)
```

Comment:
You validate a URI by parsing a URL (and to make it more complicated: a java  based URL, which only knows about specific protocols and schemes).
A valid URI is: `/etc/docker` or `hdfs://namenode-host-name:web-interface-port/data/file-path`
which both fail on my machine.
You could use to parse a URI (http://www.ietf.org/rfc/rfc2396.txt) instead.